src/core/lib/event_engine/thread_pool/thread_pool_factory.cc

```diff
@@ -16,19 +16,73 @@
 #include <stddef.h>
 
 #include <memory>
+#include <utility>
+#include <vector>
 
 #include <grpc/support/cpu.h>
 
+#include "src/core/lib/event_engine/forkable.h"
 #include "src/core/lib/event_engine/thread_pool/thread_pool.h"
 #include "src/core/lib/event_engine/thread_pool/work_stealing_thread_pool.h"
 #include "src/core/lib/gpr/useful.h"
+#include "src/core/lib/gprpp/no_destruct.h"
 
 namespace grpc_event_engine {
 namespace experimental {
 
+namespace {
+
+grpc_core::NoDestruct<std::vector<std::weak_ptr<ForkableInterface>>>
+    g_forkable_instances;
+
+bool g_registered{false};
```

Comment:
This can live inside the pthread guards.

Reply:
Done.